apps/web/app/environments/[environmentId]/surveys/templates/PreviewSurveyServer.tsx

```diff
@@ -0,0 +1,327 @@
+import FormbricksSignature from "@/components/preview/FormbricksSignature";
+import Modal from "@/components/preview/Modal";
+import Progress from "@/components/preview/Progress";
+import QuestionConditional from "@/components/preview/QuestionConditional";
+import ThankYouCard from "@/components/preview/ThankYouCard";
+// import { useEnvironment } from "@/lib/environments/environments";
+// import { useProduct } from "@/lib/products/products";
+import type { Logic, Question } from "@formbricks/types/questions";
+import { Survey } from "@formbricks/types/surveys";
+import { useEffect, useRef, useState } from "react";
+import type { TProduct } from "@formbricks/types/v1/product";
+import type { TEnvironment } from "@formbricks/types/v1/environment";
+interface PreviewSurveyProps {
+  setActiveQuestionId: (id: string | null) => void;
+  activeQuestionId?: string | null;
+  questions: Question[];
+  brandColor: string;
+  environmentId: string;
+  surveyType: Survey["type"];
+  thankYouCard: Survey["thankYouCard"];
+  autoClose: Survey["autoClose"];
+  previewType?: "modal" | "fullwidth" | "email";
+  product: TProduct;
+  environment: TEnvironment;
+}
+
+export default function PreviewSurvey({
+  setActiveQuestionId,
+  activeQuestionId,
+  questions,
+  brandColor,
+  // environmentId,
+  surveyType,
+  thankYouCard,
+  autoClose,
+  previewType,
+  product,
+  environment,
+}: PreviewSurveyProps) {
+  const [isModalOpen, setIsModalOpen] = useState(true);
```

Comment:
I'm a bit confused by the filename `PreviewSurveyServer.tsx` and then you use client-only methods `useState`.
If it's a client component (e.g. using `useState` or `useEffect`) it must have `'use client'` at the top of the file.

Reply:
And it seems this is nearly a copy of `apps/web/app/environments/[environmentId]/surveys/PreviewSurvey.tsx`.
I think it doesn't make sense to have both files. What is the difference in this file and how can we only have one `PreviewSurvey` component?

Reply:
The major difference between them is: product, environment is received as props in `PreviewSurveyServer.tsx` instead of fetching through useSWR. This previewSurvey component is used in a bunch of places hence didn't want to make the changes of props everywhere, thus made a copy and named it as PreviewSurveyServer to signal that it can be used in server-component.